ding2_v4.vcl

```diff
@@ -0,0 +1,183 @@
+    set req.http.Cookie = regsuball(req.http.Cookie, ";(SESS[a-z0-9]+|NO_CACHE)=", "; \1=");
+    set req.http.Cookie = regsuball(req.http.Cookie, ";[^ ][^;]*", "");
+    set req.http.Cookie = regsuball(req.http.Cookie, "^[; ]+|[; ]+$", "");
+
+    if (req.http.Cookie == "") {
+      # If there are no remaining cookies, remove the cookie header. If there
+      # aren't any cookie headers, Varnish's default behavior will be to cache
+      # the page.
+      unset req.http.Cookie;
+    }
+    else {
+      # If there are any cookies left (a session or NO_CACHE cookie), do not
+      # cache the page. Pass it on to Apache directly.
+      return (pass);
+    }
+  }
+
+  # Handle compression correctly. Different browsers send different
+  # "Accept-Encoding" headers, even though they mostly all support the same
+  # compression mechanisms. By consolidating these compression headers into
+  # a consistent format, we can reduce the size of the cache and get more hits.
+  # @see: http:// varnish.projects.linpro.no/wiki/FAQ/Compression
+  if (req.http.Accept-Encoding) {
+    if (req.http.Accept-Encoding ~ "gzip") {
+      # If the browser supports it, we'll use gzip.
+      set req.http.Accept-Encoding = "gzip";
+    }
+    else if (req.http.Accept-Encoding ~ "deflate") {
+      # Next, try deflate if it is supported.
+      set req.http.Accept-Encoding = "deflate";
+    }
+    else {
+      # Unknown algorithm. Remove it and send unencoded.
+      unset req.http.Accept-Encoding;
+    }
+  }
+  return (hash);
+}
+
+sub vcl_deliver {
+  # If the response contains the X-Drupal-Roles header and the request URL
+  # is right. Copy the X-Drupal-Roles header over to the request and restart.
+  if (req.url == "/varnish/roles" && resp.http.X-Drupal-Roles) {
+    set req.http.X-Drupal-Roles = resp.http.X-Drupal-Roles;
+    set req.url = req.http.X-Original-URL;
+    unset req.http.X-Original-URL;
+    return (restart);
+  }
+
+  # If responces X-Drupal-Roles is not set, move it from the request.
+  if (!resp.http.X-Drupal-Roles) {
+    set resp.http.X-Drupal-Roles = req.http.X-Drupal-Roles;
+  }
+
+  # Remove server information
+  set resp.http.X-Powered-By = "Ding T!NG";
+
+  # Debug
+  if (obj.hits > 0 ) {
+    set resp.http.X-Varnish-Cache = "HIT";
+  }
+  else {
+    set resp.http.X-Varnish-Cache = "MISS";
+  }
+}
+
+# Code determining what to do when serving items from the Apache servers.
+sub vcl_backend_response {
+  # Allow items to be stale if needed.
+  set beresp.grace = 6h;
+
+  # We need this to cache 404s, 301s, 500s. Otherwise, depending on backend but
+  # definitely in Drupal's case these responses are not cacheable by default.
+  if (beresp.status == 404 || beresp.status == 301 || beresp.status == 500) {
+    set beresp.ttl = 10m;
+  }
+
+  if (beresp.http.x-drupal-cache == "HIT") {
+    unset beresp.http.Cache-Control;
+    set beresp.http.Cache-Control = "public,max-age=21600";
+  }
```

Comment:
This always seems to cache pages which are also in Drupals cache with a hardcoded max-age. Why is this?

Reply:
@cableman: Where do you see this in [the existing `ding2.vcl`](https://github.com/ding2/ding2/blob/master/ding2.vcl)?

Reply:
Right removed it... can't remember why!